openmc/mgxs_library.py

```diff
@@ -0,0 +1,2577 @@
+
+        See also
+        --------
+        openmc.mgxs.Library.create_mg_library()
+        openmc.mgxs.Library.get_xsdata()
+
+        """
+
+        check_type('fission', fission, openmc.mgxs.FissionXS)
+        check_value('energy_groups', fission.energy_groups,
+                    [self.energy_groups])
+        check_value('domain_type', fission.domain_type,
+                    openmc.mgxs.DOMAIN_TYPES)
+        check_type('temperature', temperature, Real)
+        check_value('temperature', temperature, self.temperatures)
+
+        i = np.where(self.temperatures == temperature)[0][0]
+        self._fission[i] = fission.get_xs(nuclides=nuclide,
+                                          xs_type=xs_type,
+                                          subdomains=subdomain)
+
+    def set_nu_fission_mgxs(self, nu_fission, temperature=294.,
+                            nuclide='total', xs_type='macro', subdomain=None):
+        """This method allows for an openmc.mgxs.FissionXS
+        to be used to set the nu-fission cross section for this XSdata object.
+
+        Parameters
+        ----------
+        nu_fission: openmc.mgxs.FissionXS
+            MGXS Object containing the nu-fission cross section
+            for the domain of interest.
+        temperature : float
+            Temperature (in Kelvin) of the data. Defaults to room temperature
+            (294K).
+        nuclide : str
+            Individual nuclide (or 'total' if obtaining material-wise data)
+            to gather data for.  Defaults to 'total'.
+        xs_type: {'macro', 'micro'}
+            Provide the macro or micro cross section in units of cm^-1 or
+            barns. Defaults to 'macro'.
+        subdomain : iterable of int
+            If the MGXS contains a mesh domain type, the subdomain parameter
+            specifies which mesh cell (i.e., [i, j, k] index) to use.
+
+        See also
+        --------
+        openmc.mgxs.Library.create_mg_library()
+        openmc.mgxs.Library.get_xsdata()
+
+        """
+
+        check_type('nu_fission', nu_fission, (openmc.mgxs.FissionXS,
+                                              openmc.mgxs.NuFissionMatrixXS))
+        if isinstance(nu_fission, openmc.mgxs.FissionXS):
+            check_value('nu', nu_fission.nu, [True])
+        check_value('energy_groups', nu_fission.energy_groups,
+                    [self.energy_groups])
+        check_value('domain_type', nu_fission.domain_type,
+                    openmc.mgxs.DOMAIN_TYPES)
+        check_type('temperature', temperature, Real)
+        check_value('temperature', temperature, self.temperatures)
+
+        i = np.where(self.temperatures == temperature)[0][0]
+        self._nu_fission[i] = nu_fission.get_xs(nuclides=nuclide,
+                                                xs_type=xs_type,
+                                                subdomains=subdomain)
+
+        if np.sum(self._nu_fission) > 0.0:
+            self._fissionable = True
```

Comment:
Good candidate for standalone method, since there are 8 of these `fissionable` checks.

```
def set_fissionable(fission_type):
      if np.sum(fission_type) > 0.0:
            self._fissionable = True

```

Then in the code:

```
self.set_fissionable(self._nu_fission)
```